### ISSUE TITLE
get_fragment_spreads IndexError: pop from empty list

### ISSUE THREAD

**Issue by Bingdom** [Nuitka/Nuitka#2052]
## Python Version

* Please check out if the develop version of Nuitka works better for you.

Same issue for the develop version. **It works with Nuitka v1.2.2**

  > python -m nuitka --version

```
1.4.6
Commercial: None
Python: 3.10.8 (main, Nov 24 2022, 13:21:08) [GCC 10.2.1 20210110]
Flavor: Unknown
Executable: /usr/local/bin/python3.10
OS: Linux
Arch: x86_64
Distribution: Debian 11
Version C compiler: /usr/bin/gcc (gcc).
```

## Error output

```
Error handling request
Traceback (most recent call last):
  File "/home/name/test.dist/aiohttp/web_protocol.py", line 433, in _handle_request
  File "/home/name/test.dist/aiohttp/web_app.py", line 504, in _handle
  File "/home/name/test.dist/strawberry/aiohttp/views.py", line 79, in __call__
  File "/home/name/test.dist/strawberry/aiohttp/handlers/http_handler.py", line 41, in handle
  File "/home/name/test.dist/strawberry/aiohttp/handlers/http_handler.py", line 66, in post
  File "/home/name/test.dist/strawberry/aiohttp/handlers/http_handler.py", line 87, in execute_request
  File "/home/name/test.dist/strawberry/schema/schema.py", line 226, in execute
  File "/home/name/test.dist/strawberry/schema/execute.py", line 112, in execute
  File "/home/name/test.dist/strawberry/schema/execute.py", line 53, in _run_validation
  File "/home/name/test.dist/strawberry/schema/execute.py", line 41, in validate_document
  File "/home/name/test.dist/graphql/validation/validate.py", line 86, in validate
  File "/home/name/test.dist/graphql/language/visitor.py", line 261, in visit
  File "/home/name/test.dist/graphql/utilities/type_info.py", line 319, in leave
  File "/home/name/test.dist/graphql/language/visitor.py", line 357, in leave
  File "/home/name/test.dist/graphql/validation/rules/no_undefined_variables.py", line 29, in leave_operation_definition
  File "/home/name/test.dist/graphql/validation/validation_context.py", line 223, in get_recursive_variable_usages
  File "/home/name/test.dist/graphql/validation/validation_context.py", line 148, in get_recursively_referenced_fragments
  File "/home/name/test.dist/graphql/validation/validation_context.py", line 119, in get_fragment_spreads
IndexError: pop from empty list
```

* How did you install Nuitka and Python

Built Python myself. Installed Nuitka via pip.

* The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g.
  `numpy`, you are expected to shorten this to the relevant ones.

  > python -m pip freeze

```
aiohttp==3.8.3
strawberry-graphql==0.155.4
```

* Also supply a [Short, Self Contained, Correct, Example](https://sscce.org/)

```
import strawberry
from aiohttp import web
from strawberry.aiohttp.views import GraphQLView

# Available queries
# GraphQL (Not working)
@strawberry.type
class Query:
    @strawberry.field
    def ping() -> bool:
        return True

# REST (Working)
async def handle(request):
    return web.Response(text="I'm Alive!")

view = GraphQLView(schema=strawberry.Schema(query=Query))

app = web.Application()
app.add_routes([
    web.post('/graphql', view),
    web.post('/ping', handle)
])

if __name__ == "__main__":
    web.run_app(app)
```

Please try the following requests. HTTP is successful, but the GraphQL request is not, and the error above is outputted

* Provide in your issue the Nuitka options used

```
--standalone
```

* Note if this is a regression

It works completely fine at v1.2.2, but I'm not sure of any other versions after that.

**Comment by kayhayen** [comment]
I am getting `405: Method Not Allowed` with Python and Nuitka alike, I don't think I know what to post there, so I doubt I can reproduce this, switching to `get` gives an error, and probably won't exercise the code, or will it.

**Comment by Bingdom** [comment]
Hi Kay,

This example uses the POST request. Here's 2 curl requests that you can use to reproduce

HTTP:
```bash
curl -X POST http://localhost:8080/ping
```
GraphQL (Not working in Krita):
```bash
curl -X POST http://localhost:8080/graphql -d '{"query": "query {ping}"}'
```

**Comment by kayhayen** [comment]
Ok, that did it for me.

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback if this is working is very welcome, just please do not share plans of doing it, but rather confirmations or denials of it working.

**Comment by kayhayen** [comment]
This is part of the hotfix release 1.4.7 that I just made.
